packages/core-utils/src/route.js

```diff
@@ -1,31 +1,151 @@
 /**
- * Looks up an operator from the provided configuration.
- * @param  {string} feedId - the feed ID, which can be obtained by splitting the
- *                            OTP route or stop ID by `:`
- * @param  {object} transitOperators - transitOperators from config.
- * @return {object} the operator for the specified feedId
+ * Returns the transit operator (if an exact match is found) from the transit
+ * operators config value. It is critical to use both the feedId and agencyId in
+ * this method because it is possible in OTP for there to be a duplicate
+ * agencyId in separate feeds.
+ *
+ * @param  {string} feedId The feedId that this transit agency belongs to
+ * @param  {string} agencyId The agencyId of the transit agency
+ * @param  {array} transitOperators The transitOperators list from the config
+ * @return {object} The transitOperator if a match was found or null if no match
+ *    was found
  */
-export function getTransitOperatorFromId(feedId, transitOperators) {
+export function getTransitOperatorFromFeedIdAndAgencyId(
+  feedId,
+  agencyId,
+  transitOperators
+) {
   return (
-    transitOperators.find(transitOperator => transitOperator.id === feedId) ||
-    null
+    transitOperators.find(
+      transitOperator =>
+        transitOperator.feedId === feedId &&
+        transitOperator.agencyId === agencyId
```

Comment:
There is a way to do this, but since we don't auto-generate docs, I'm not sure it adds much value. The JSDoc comment already says that this is a value from the transitOperators list in the config.